docs/modules/ROOT/pages/patterns.adoc

```diff
@@ -0,0 +1,216 @@
+
+```javascript
+const pattern = new Cypher.Pattern(person)
+            .withProperties({
+                name: new Cypher.Param("Person")
+            })
+            .related(actedIn)
+            .to(movie);
+```
+
+```cypher
+(this0:Person { name: $param0 })-[this1:ACTED_IN]->(this2:Movie)
+```
+
+`withProperties` takes an object with the properties to match and the Param objects to be used as the expected values of the pattern. `withProperties` can be used in both nodes and relationship elements.
+
+=== Longer patterns
+
+Patterns are not limited to a node or a relationship. Patterns can be arbitrarily long. For example:
+
+```javascript
+const user = new Cypher.Node({labels: ["User"]});
+
+const pattern = new Cypher.Pattern(person).related(actedIn).to(movie).related().withDirection("left").to(user);
+```
+
+```cypher
+(this0:Person)-[this1:ACTED_IN]->(this2:Movie)<-[this3]-(this4:User)
+```
+
+== Advanced patterns
+
+=== Cycles
+
+A pattern may have cycles, to achieve this, we can reuse the same variables:
+
+```javascript
+const actor = new Cypher.Node({labels: ["Actor"]});
+const movie = new Cypher.Node({labels: ["Movie"]});
+
+const actedIn = new Cypher.Relationship({type: "ACTED_IN"});
+const directed = new Cypher.Relationship({type: "DIRECTED"});
+
+const pattern = new Cypher.Pattern(person).related(actedIn).to(movie).related(directed).withDirection("undirected").to(actor).withoutLabels();
+```
+
+This translates to:
+
+```cypher
+(this0:Person)-[this1:ACTED_IN]->(this2:Movie)-[this3:DIRECTED]-(this0)
+```
+
+Note how the initial node in the pattern (`this0`) is the same as the one referenced in the last element. This will match actors that also directed the same movie.
+
+=== Length
+
+The length (or hops) of a relationship can be defined with the `withLength` method.
+
+==== Exact length
+
+The exact length can be defined by passing a number:
+
+```javascript
+const pattern = new Cypher.Pattern(person).related(actedIn).withLength(3).to(movie);
+```
+
+```cypher
+MATCH (this0:Person)-[this1:ACTED_IN*3]->(this2:Movie)
+```
+
+==== Min and Max length
+
+Bounds can be added by passing an object with the following options:
+
+* `min` (optional): Define the minimum length of the relationship.
+* `max` (optional): Define the maximum length of the relationship.
+
+```javascript
+const pattern = new Cypher.Pattern(person).related(actedIn).withLength({min: 2, max: 10}).to(movie);
+```
+
+```cypher
+MATCH (this0:Person)-[this1:ACTED_IN*2..10]->(this2:Movie)
+```
+
+==== Any length
+
+By using the string `"*"`, a relationship with any length will be matched:
+
+```javascript
+const pattern = new Cypher.Pattern(person).related(actedIn).withLength("*").to(movie);
+```
+
+```cypher
+MATCH (this0:Person)-[this1:ACTED_IN*]->(this2:Movie)
+```
+
+== Partial patterns
+
+As mentioned, all patterns begin and end with a Node, however, it is possible to define a _Partial Pattern_ by using `.related` without `.to` afterwards:
```

Comment:
```suggestion
All patterns begin and end with a Node, however, it is possible to define a _Partial Pattern_ by using `.related` without `.to` afterwards:
```

I'm not 100% sure, but I think it's more correct to not assume another part of the docs was read.